honestcash-v2/src/app/auth/services/auth.service.ts

```diff
@@ -162,13 +103,13 @@ export class AuthService {
 
   public changePassword(context: ResetPasswordContext): Observable<OkResponse> {
     return defer(async () => {
-      const mnemonic = (await WalletUtils.generateNewWallet(context.newPassword)).mnemonic;
-      const mnemonicEncrypted = await WalletUtils.encrypt(mnemonic, context.newPassword);
+      const mnemonic = (await this.walletService.createWallet(context.newPassword)).mnemonic;
```

Comment:
these can also probably be static functions (createWallet), I will check possibility of further refactoring